File: sdks/dotnet/tests/fixtures/fake_llmux.py
#!/usr/bin/env python3
"""Fake llmux binary for .NET SDK tests.

Honors LLMUX_ADDR=127.0.0.1:<port>, serves GET /health -> 200.
  FAKE_HEALTH_STATUS  status for /health (default 200)
  FAKE_NEVER_LISTEN   if "1", never binds (simulates a hung start)
"""
import os
import signal
import sys
import time
from http.server import BaseHTTPRequestHandler, HTTPServer


def main():
    signal.signal(signal.SIGTERM, lambda *_: os._exit(0))
    if os.environ.get("FAKE_NEVER_LISTEN") == "1":
        time.sleep(30)
        return
    host, _, port = os.environ.get("LLMUX_ADDR", "127.0.0.1:0").partition(":")
    status = int(os.environ.get("FAKE_HEALTH_STATUS", "200"))

    class H(BaseHTTPRequestHandler):
        def do_GET(self):
            if self.path == "/health":
                self.send_response(status)
                self.end_headers()
                self.wfile.write(b"ok")
            else:
                self.send_response(404)
                self.end_headers()

        def log_message(self, *a):
            pass

    HTTPServer((host, int(port)), H).serve_forever()


if __name__ == "__main__":
    sys.exit(main())

File: sdks/ruby/test/fixtures/fake_llmux.rb
#!/usr/bin/env ruby
# frozen_string_literal: true

# Test fixture: a fake llmux binary.
# Honors LLMUX_ADDR=127.0.0.1:<port>, binds it, serves GET /health -> 200.
#   FAKE_HEALTH_STATUS  status for /health (default 200)
#   FAKE_NEVER_LISTEN   if "1", never binds (simulates a hung start)

require "socket"

trap("TERM") { exit 0 }
trap("INT") { exit 0 }

if ENV["FAKE_NEVER_LISTEN"] == "1"
  sleep 30
  exit 0
end

addr = ENV["LLMUX_ADDR"] || "127.0.0.1:0"
host, port = addr.split(":")
status = (ENV["FAKE_HEALTH_STATUS"] || "200").to_i
reason = status == 200 ? "OK" : "Unavailable"

server = TCPServer.new(host, port.to_i)
loop do
  conn = server.accept
  request_line = conn.gets.to_s
  # Drain headers.
  while (line = conn.gets) && line != "\r\n"; end
  if request_line.start_with?("GET /health")
    conn.write "HTTP/1.1 #{status} #{reason}\r\nContent-Length: 2\r\nConnection: close\r\n\r\nok"
  else
    conn.write "HTTP/1.1 404 Not Found\r\nContent-Length: 0\r\nConnection: close\r\n\r\n"
  end
  conn.close
rescue StandardError
  conn&.close
end
